pkg/util/utils.go

```diff
@@ -303,6 +306,513 @@ func GetNoMapMapping() (*stypes.IDMappingOptions, int, int, error) {
+//
+// Flags:
+// ======
+// Flags can be used to tell the caller how should the mapping be interpreted
+//
+// The `u` and `g` flags can be used to enforce that the mapping applies
+// exclusively to UIDs or GIDs.
+//
+// The `+` flag is interpreted as if the mapping replaces previous mappings
+// removing any conflicting mapping from those before adding this one.
+func parseTriple(spec []string, parentMapping []ruser.IDMap, mapSetting string) (mapping idtools.IDMap, flags idMapFlags, err error) {
+	if len(spec[0]) == 0 {
+		return mapping, flags, fmt.Errorf("invalid empty container id at %s map: %v", mapSetting, spec)
+	}
+	var cid, hid uint64
+	var forceSizeOne bool
+	flags.Extends = false
+	flags.UserMap = false
+	flags.GroupMap = false
+	var i = 0
+FlagDetector:
+	for index, char := range spec[0] {
+		i = index
+		switch {
+		case '0' <= char && char <= '9':
+			break FlagDetector
+		case char == '+':
+			flags.Extends = true
+		case char == 'u':
+			flags.UserMap = true
+		case char == 'g':
+			flags.GroupMap = true
+		case true:
+			return mapping, flags, fmt.Errorf("invalid mapping: %v. Unknown flag %v", spec, char)
+		}
+	}
+	// If no "u" nor "g" flag is given, assume the mapping applies to both
+	if !flags.UserMap && !flags.GroupMap {
+		flags.UserMap = true
+		flags.GroupMap = true
+	}
+	// Parse the container ID, which must be an integer:
+	cid, err = strconv.ParseUint(spec[0][i:], 10, 32)
+	if err != nil {
+		return mapping, flags, fmt.Errorf("parsing id map value %q: %w", spec[0], err)
+	}
+	// Parse the host id, which may be integer or @<integer>
+	if len(spec[1]) == 0 {
+		return mapping, flags, fmt.Errorf("invalid empty host id at %s map: %v", mapSetting, spec)
+	}
+	if spec[1][0] != '@' {
+		forceSizeOne = false
+		hid, err = strconv.ParseUint(spec[1], 10, 32)
+	} else {
+		// Parse @<id>, where <id> is an integer corresponding to the parent mapping
+		// TODO: Accept @<name>, where <name> is a string corresponding to a parent user/group name
+		forceSizeOne = true
+		hparentid, err := strconv.ParseUint(spec[1][1:], 10, 32)
+		if err != nil {
+			return mapping, flags, fmt.Errorf("parsing id map value %q: %w", spec[1], err)
+		}
+		// Map the parent mapping to this one, to get the actual value:
+		if (mapSetting == "UID" && flags.UserMap) || (mapSetting == "GID" && flags.GroupMap) {
+			hid, err = mapIDwithMapping(hparentid, parentMapping, mapSetting)
+			if err != nil {
+				return mapping, flags, err
+			}
+		} else {
+			hid = 0
+		}
+	}
+	if err != nil {
+		return mapping, flags, fmt.Errorf("parsing id map value %q: %w", spec[1], err)
+	}
+	// Parse the size of the mapping, which must be an integer
+	sz, err := strconv.ParseUint(spec[2], 10, 32)
+	if err != nil {
+		return mapping, flags, fmt.Errorf("parsing id map value %q: %w", spec[2], err)
+	}
+	// When using @<id>, the size must be one. Otherwise the semantic is confusing
+	// (should it extend at the current mapping or at the parent mapping?)
```

Comment:
I think current mapping is less confusing

Reply:
I think the behaviour described in this other review comment makes sense: https://github.com/containers/podman/pull/18713#discussion_r1301264823